crawl-ref/source/package.cc

```diff
@@ -676,7 +676,9 @@ plen_t package::get_chunk_compressed_length(const string &name)
 chunk_writer::chunk_writer(package *parent, const string &_name)
     : first_block(0), cur_block(0), block_len(0)
 {
+#ifndef __ANDROID__
```

Comment:
conditionally defining this assert seems iffy, since if it fails subsequent code will probably segfault...

If it's really ok for this to be called with `nullptr` parent, probably it should have an early return or following code (e.g. `pkg` on line 690) needs something else

Reply:
You are right, sorry. I shouldn't have accepted the first solution I found for the crash logs. After further testing I found the root cause of the problem. New solution in 6d99694.